<?xml version="1.0" encoding="UTF-8"?>

<package name="javax.security.sasl" release="1">
    <class name="SaslException">
        <constructor>
            <RegionEffects>none</RegionEffects>
            <Starts>nothing</Starts>
            <Unique>return</Unique>
        </constructor>

        <constructor params="java.lang.String">
            <RegionEffects>none</RegionEffects>
            <Starts>nothing</Starts>
            <Unique>return</Unique>
        </constructor>

        <constructor params="java.lang.String, java.lang.Throwable">
            <RegionEffects>none</RegionEffects>
            <Starts>nothing</Starts>
            <Unique>return</Unique>
        </constructor>

        <method name="getCause">
            <Borrowed>this</Borrowed>
            <RegionEffects>reads Instance</RegionEffects>
            <Starts>nothing</Starts>
        </method>

        <method name="initCause" params="java.lang.Throwable">
            <RegionEffects>writes Instance</RegionEffects>
            <Starts>nothing</Starts>
        </method>

        <method name="toString">
            <Borrowed>this</Borrowed>
            <RegionEffects>reads All</RegionEffects>
            <Starts>nothing</Starts>
        </method>
    </class>
</package>
